server/aap/publish/formatters/aap_apple_news_formatter.py

```diff
@@ -52,44 +52,123 @@ def format(self, article, subscriber, codes=None):
         except Exception as ex:
             raise AppleNewsError.AppleNewsFormatter(exception=ex)
 
-    def _format(self, article):
-        apple_news = {}
-        self._parse_content(article)
-        if not article.get('_title') or not article.get('_analysis_first_line') or not article.get('_analysis') \
-            or not article.get('_statement') or not article.get('_statement_attribution') or \
-                not article.get('_verdict1') or not article.get('_verdict2') or not article.get('_references'):
-            missing_fields = {
-                'title': True if article.get('_title') else False,
-                'subtitle': True if article.get('_analysis_first_line') else False,
-                'analysis': True if article.get('_analysis') else False,
-                'statement': True if article.get('_statement') else False,
-                'statement_attribution': True if article.get('_statement_attribution') else False,
-                'verdict1': True if article.get('_verdict1') else False,
-                'verdict2': True if article.get('_verdict2') else False,
-                'references': True if article.get('_references') else False,
-            }
+    def _filter_blocks(self, item, field, bfilter, remove):
+        """
+        Function to filter the embed blocks for video and audio and also will regenerate the html in a more friendly
+        form using the AppleExporter class
+        :param item: The article
+        :param field: the field to operate on
+        :param bfilter: Filter function to determine if the block is to be kept
+        :param remove: list of keys to remove
+        :return:
+        """
+        editor = Editor3Content(item, field, True)
+        exporter = AppleExporter(editor)
+        editor.html_exporter = exporter
+        blocks = []
+        for block in editor.blocks:
+            if bfilter(block, remove):
+                blocks.append(block)
+        editor.set_blocks(blocks)
+        editor.update_item()
+
+    def _not_embed(self, block, remove):
+        if block.type.lower() == "atomic":
+            bk = [e.key for e in block.entities if e.key in remove]
+            if bk:
+                return False
+        return True
+
+    def _remove_embeds(self, article, remove_keys):
+        """
+        Removes the nominated embeds from the draftjs state and regenerates the HTML.
+        :param article:
+        :param remove_keys
+        :return:
+        """
+        to_remove = [k.lstrip("editor_") for k in remove_keys]
+        fields = get_content_state_fields(article)
+        for field in fields:
+            self._filter_blocks(article, field, self._not_embed, to_remove)
+
+        for key in remove_keys:
+            article.get("associations", {}).pop(key, None)
+            if "refs" in article:
```

Comment:
In case `refs` is in article but have a value of `None`, it might help to specifically test for this, i.e. `if article.get("refs") is not None`